lib/components/isometric_tile_map_component.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+
+import '../flame.dart';
+import '../position.dart';
+import '../sprite.dart';
+
+class IsometricTileset {
+  final Image tileset;
+  final int size;
+
+  final Map<int, Sprite> _spriteCache = {};
+
+  IsometricTileset(this.tileset, this.size);
+
+  int get cols => tileset.width ~/ size;
+
+  Sprite getTile(int tileId) {
+    return _spriteCache[tileId] ??= _computeTile(tileId);
+  }
+
+  Sprite _computeTile(int tileId) {
+    final i = tileId % cols;
+    final j = tileId ~/ cols;
+    final s = size.toDouble();
+    return Sprite.fromImage(tileset, x: s * i, y: s * j, width: s, height: s);
+  }
+
+  static Future<IsometricTileset> load(String fileName, int size) async {
+    final image = await Flame.images.load(fileName);
+    return IsometricTileset(image, size);
+  }
+}
+
+class Block {
+  int x, y;
+  Block(this.x, this.y);
+
+  @override
+  String toString() => '($x, $y)';
+}
+
+class IsometricTileMapComponent extends PositionComponent {
+  IsometricTileset tileset;
+  List<List<int>> matrix;
+  int destTileSize;
```

Comment:
times shouldn't be double? as most of sizes and positions on flutter and flame are doubles?

Reply:
yeah I was debating that, because it feels really weird to be a double haha

Reply:
in fact we already use in on spritesheet
```
  int textureWidth;
  int textureHeight;
```